cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsDownscaleService.java

```diff
@@ -217,29 +218,35 @@ private int calculateAutoScalingGroupSize(List<String> instanceIdsInAutoScalingG
     }
 
     private List<String> terminateInstances(Long stackId, List<String> instanceIdsToDelete, AmazonEc2Client amazonEC2Client) {
-        LOGGER.debug("Terminated instances. [stack: {}, instances: {}]", stackId, instanceIdsToDelete);
-        try {
-            List<String> existingInstances = getExistingInstances(instanceIdsToDelete, amazonEC2Client);
-            if (!existingInstances.isEmpty()) {
-                amazonEC2Client.terminateInstances(new TerminateInstancesRequest().withInstanceIds(existingInstances));
-            }
-            return existingInstances;
-        } catch (AmazonServiceException e) {
-            LOGGER.info("Termination failed, lets check if it is because instance was not found", e);
-            if (!INSTANCE_NOT_FOUND_ERROR_CODE.equals(e.getErrorCode())) {
-                throw e;
-            } else {
-                LOGGER.info("Instance was not found, lets terminate others");
-                List<String> runningInstances = instanceIdsToDelete.stream()
-                        .filter(instanceId -> !e.getMessage().contains(instanceId))
-                        .collect(Collectors.toList());
-                LOGGER.info("Running instances on AWS to terminate: {}", runningInstances);
-                if (!runningInstances.isEmpty()) {
-                    amazonEC2Client.terminateInstances(new TerminateInstancesRequest().withInstanceIds(runningInstances));
+        while (instanceIdsToDelete.size() > 0) {
+            int originalDeletableInstanceIdsSize = instanceIdsToDelete.size();
```

Comment:
this should be before the loop

Reply:
oh, I think I misunderstood the purpose of this as you reused the method parameter